src/com/sun/jna/NativeLibrary.java

```diff
@@ -132,7 +138,7 @@ private static NativeLibrary loadLibrary(String libraryName, Map options) {
 	}
 
         boolean isAbsolutePath = new File(libraryName).isAbsolute();
-        List searchPath = new LinkedList();
+        List<String> searchPath = new LinkedList<String>();
```

Comment:
A _LinkedList_ is not really a good _List_ since it has _O(n)_ access time. If you expect random access via _List#get(int)_ then use an _ArrayList_. On the other hand, if you expect to access the search path only **sequentially** then the correct definition is:
```java
Collection<String> searchPath = new LinkedList<String>();
```
This may seem like nitpicking, but many do not distinguish between a _List_ and a _Collection_, and since (unfortunately) _LinkedList_ implements _List_ there is no compile-time distinction between them. Basically, if you expect to access non-sequential locations then use a _List_ (usually an _ArrayList_), otherwise use a _Collection_ - in which case it may be a _LinkedList, HashSet, LinkedHashSet_ - depending on the expected behavior.

The same goes for input / output parameters - if you expect to process values **sequentially** then don't ask for a _List_ as the input parameter:
```java
public String findSomething(Collection<String> candidates) {
    for (String c : candidates) {
          if ...some condition... {
             return c;
          }
    }

    return null;
}
```
In the above example requiring a _List_ of candidates makes no sense since the code does not care about a specific order or non-sequential access.